pkg/actuators/machine/actuator.go

```diff
@@ -421,7 +421,7 @@ func (a *Actuator) Update(context context.Context, cluster *clusterv1.Cluster, m
 		return a.handleMachineError(machine, err, updateEventAction)
 	}
 	// Get all instances not terminated.
-	existingInstances, err := a.getMachineInstances(cluster, machine)
+	existingInstances, err := getExistingInstances(machine, client)
```

Comment:
This is to ensure we check existence consistently everywhere and we look for instanceID before falling back to tags